pkg/find/find.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2015-2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package find
+
+import (
+	"os"
+	"path/filepath"
+	"regexp"
+)
+
+type Names struct {
+	Name string
+	os.FileInfo
+}
+
+type Finder struct {
+	Root     string
+	Pattern  string
+	Match    *regexp.Regexp
+	Mode     os.FileMode
+	ModeMask os.FileMode
+	Debug    func(string, ...interface{})
+	Name     chan *Names
+	Err      chan error
+}
+
+type Set func(f *Finder) error
+
+func New(opts ...Set) (*Finder, error) {
+	// All of these can be overridden by the opts
+	var f = &Finder{Root: "/", Pattern: ".*"}
+	f.Debug = func(string, ...interface{}) {}
+	f.Name = make(chan *Names, 128)
+	f.Err = make(chan error, 128)
+	for _, opt := range opts {
+		if err := opt(f); err != nil {
+			return nil, err
+		}
+	}
+	var err error
+	f.Match, err = regexp.Compile(f.Pattern)
+	f.Debug("Create new Finder: %v", f)
+	return f, err
+}
+
+func (f *Finder) Find() {
+	filepath.Walk(f.Root, func(n string, fi os.FileInfo, err error) error {
+		if err != nil {
+			f.Err <- err
+			return nil
+		}
+		// If it matches, then push its name into the result channel,
+		// and keep looking.
+		if !f.Match.Match([]byte(n)) {
```

Comment:
Skip this match conditional if name is an empty string.

Reply:
done